src/format/mod.rs

```diff
@@ -638,6 +672,41 @@ fn format_inner<'a>(
                     }
                 };
 
+            match ret {
+                Some(ret) => ret?,
+                None => return Err(fmt::Error), // insufficient arguments for given format
+            }
+        },
+
+        Item::FixedExt(ref spec) => {
+            use self::FixedExt::*;
+
+            let ret = match spec {
+                &OneLetterMonthName =>
+                    date.map(|d| {
+                        result.push_str(ONE_LETTER_MONTHS[d.month0() as usize]);
+                        Ok(())
+                    }),
+                &OneLetterWeekdayName =>
+                    date.map(|d| {
+                        result.push_str(
+                            ONE_LETTER_WEEKDAYS[d.weekday().num_days_from_monday() as usize]
+                        );
+                        Ok(())
+                    }),
+                &OneLetterLowerAmPm =>
+                    time.map(|t| {
+                        result.push_str(if t.hour12().0 {"p"} else {"a"});
+                        Ok(())
+                    }),
+                &OneLetterUpperAmPm =>
+                    time.map(|t| {
+                        result.push_str(if t.hour12().0 {"P"} else {"A"});
+                        Ok(())
+                    }),
+                _ => return Err(fmt::Error),
```

Comment:
Could you explicitly match on `__DoNotMatch`? We're doing this internally, and are guaranteed to update in lockstep, we just want downstream users to know that they shouldn't.